ttnn/cpp/ttnn/tensor/types.hpp

```diff
@@ -275,6 +275,7 @@ struct MemoryConfig {
     TensorMemoryLayout memory_layout = TensorMemoryLayout::INTERLEAVED;  // Interleave the data across multiple banks
     BufferType buffer_type = BufferType::DRAM;                           // Can be either DRAM or L1
     std::optional<ShardSpec> shard_spec = std::nullopt;
+    bool keep_l1_aligned = false;
```

Comment:
This doesn't seem like the right place for this. @sminakov-tt @ayerofieiev-tt can you provide input

Reply:
I think it's an alright place

Reply:
Honestly, as a user, I have no idea how I’d use this flag. 
Do we actually need this flag? 
There seem to be a  clear logic which we pushed to a user to follow. Why can’t we instead just follow it ourself?

Reply:
The need for this will be gone when we have unification across all ops on standardizing the source of alignment. Some ops have alignment hardcoded in their kernels based off mem type others use the buffer struct (which currently has a bug of it being hardcoded). 

This is needed to allow max pool which assumes proper alignment based off mem type to work without breaking the rest, we should follow up when we fix the buffer alignment ( @abhullar-tt has a fix but that introduces bugs for kernels that are hardcoded incorrectly as mentioned above).

Reply:
For now this is required for @mywoodstock and @wransom-TT s max pool stuff to work so I propose we push this in this PR, and then in a future PR have a push across all ops to fix alignment to use the buffer struct with @abhullar-tt 's fix.

Reply:
@ntarafdar yes we need to get this in -- and may be create an issue w.r.t. cleanup once the alignment fix is in, if not already there.

Reply:
Alright looks like we are in agreement that we need this in :) . Good fix @llongTT .

Reply:
If this is only needed for i2s and s2i ops, shouldn't this flag be there?

Reply:
> If this is only needed for i2s and s2i ops, shouldn't this flag be there?

The issue is exposed by i2s/s2i + some other ops that change the tensor shape. 
Most of the ops works fine with i2s/s2i in the buggy way, need to figure out the impact if we enforce l1 alignment